### PR TITLE
KLBP-349: Added disabled state for next button if there is an invalid…

### PR DESCRIPTION
**Pull request by iuliudascalu**
… document.

### CODE REVIEW

**Review by andreibuhus — APPROVED** [review]

**Review by andreir-suciu — APPROVED** [review]

File: src/components/Journeys/Common/SecondaryPersonViewMode.tsx [hunk 1]
```diff
@@ -57,6 +62,29 @@ const SecondaryPersonViewMode: FC<ISecondaryPersonViewSection> = ({
...
+      const indexPrevState = prevState[index]
+      const newState = [...prevState]
+      newState[index] = {
+        ...indexPrevState,
+        isDocumentInvalid,
+      }
+
+      newState.forEach((value, index1) => {
+        const idIndex = secondaryPersons.findIndex(
+          (value1, index2) => index2 === index1
```
Comment (mariuspop86) [inline src/components/Journeys/Common/SecondaryPersonViewMode.tsx:76]: `value1` => `firstValue`, `index2` => `secondIndex`

**Review by ioan-iacob-netea — APPROVED** [review]

File: src/interfaces/Step3.ts [hunk 2]
```diff
@@ -38,6 +39,7 @@ export interface IStep3PrivateCompanyHookData {
   showNewCard: boolean
   uploadHookData: IBeneficialOwnerUploadData
   validPersons: ISecondaryPersonData[]
+  setDocumentStatusList: Dispatch<SetStateAction<IDocumentInvalidList[]>>
```
Comment (mariuspop86) [inline src/interfaces/Step3.ts:42]: Please set this in the correct order

File: src/components/Journeys/Common/SecondaryPersonViewMode.tsx (hunk 1 above)
Comment (mariuspop86) [reply src/components/Journeys/Common/SecondaryPersonViewMode.tsx:76]: Better yet
```suggestion
          newState.forEach((state, stateIndex) => {
            const idIndex = secondaryPersons.findIndex(
          (secondaryPerson, secondaryPersonIndex) => secondaryPersonIndex === stateIndex
```

File: src/hooks/useStep2PrivateCompanyHook.ts [hunk 3]
```diff
@@ -66,6 +70,13 @@ const useStep2PrivateCompanyHook = () => {
     }
   }
 
+  const isNextDisabled =
```
Comment (mariuspop86) [inline src/hooks/useStep2PrivateCompanyHook.ts:73]: This is so hard to understand.

File: src/components/Journeys/Common/SecondaryPersonViewMode.tsx [hunk 4]
```diff
@@ -57,6 +62,29 @@ const SecondaryPersonViewMode: FC<ISecondaryPersonViewSection> = ({
     personTypeId
   )
 
+  useEffect(() => {
+    setDocumentStatusList((prevState: IDocumentInvalidList[]) => {
+      const indexPrevState = prevState[index]
+      const newState = [...prevState]
```
Comment (mariuspop86) [inline src/components/Journeys/Common/SecondaryPersonViewMode.tsx:68]:
```suggestion
      const newDocumentStatusList = [...prevState]
```

File: src/hooks/useStep2PrivateCompanyHook.ts (hunk 3 above)
Comment (mariuspop86) [reply src/hooks/useStep2PrivateCompanyHook.ts:73]: I would create a function like,
```
const  VallidateDocumentList = (value: IDocumentInvalidList)  => 
 documentStatusList.some(
      (value: IDocumentInvalidList) => value.isDocumentInvalid
    )

```

File: src/components/Journeys/Common/SecondaryPersonViewMode.tsx (hunk 1 above)
Comment (iuliudascalu) [reply src/components/Journeys/Common/SecondaryPersonViewMode.tsx]: Done.

File: src/hooks/useStep2PrivateCompanyHook.ts (hunk 3 above)
Comment (iuliudascalu) [reply src/hooks/useStep2PrivateCompanyHook.ts:73]: Done.

File: src/components/Journeys/Common/SecondaryPersonViewMode.tsx (hunk 4 above)
Comment (iuliudascalu) [reply src/components/Journeys/Common/SecondaryPersonViewMode.tsx]: Done.

**Review by mariuspop86 — APPROVED** [review]

**Pull request merged**